File: java/src/main/java/com/cillu/ports/RepositoryPort.java
package com.cillu.ports;

import com.cillu.core.domain.Player;
import com.cillu.core.domain.Team;

import java.util.List;

public interface RepositoryPort {
    Team saveTeam(Team team) throws Exception;
    List<Team> getTeams() throws Exception;
    Player savePlayer(Player player) throws Exception;
    List<Player> getPlayersByTeam(int teamID) throws Exception;
}

File: ports/repository_port.go
package ports

import "github.com/thecillu/exagonal-architecture/core/domain"

type RepositoryPort interface {
	GetTeams() ([]domain.Team, error)
	SaveTeam(team domain.Team) (domain.Team, error)
	SavePlayer(member domain.Player) (domain.Player, error)
	GetPlayersByTeam(team_id int64) ([]domain.Player, error)
}
